lib/stdlib/src/uri_string.erl

```diff
@@ -0,0 +1,2067 @@
+    case T =:= <<$]/utf8>> of
+        true -> H;
+        false -> Rest
+    end;
+remove_brackets(Addr) -> Addr.
+
+
+%%-------------------------------------------------------------------------
+%% Helper functions for calculating the parsed binary.
+%%-------------------------------------------------------------------------
+-spec calculate_parsed_scheme(binary(), binary()) -> binary().
+calculate_parsed_scheme(Input, <<>>) ->
+    strip_last_char(Input, [$:]);
+calculate_parsed_scheme(Input, Unparsed) ->
+    get_parsed_binary(Input, Unparsed).
+
+
+-spec calculate_parsed_part(binary(), binary()) -> binary().
+calculate_parsed_part(Input, <<>>) ->
+    strip_last_char(Input, [$?,$#]);
+calculate_parsed_part(Input, Unparsed) ->
+    get_parsed_binary(Input, Unparsed).
+
+
+-spec calculate_parsed_userinfo(binary(), binary()) -> binary().
+calculate_parsed_userinfo(Input, <<>>) ->
+    strip_last_char(Input, [$?,$#,$@]);
+calculate_parsed_userinfo(Input, Unparsed) ->
+    get_parsed_binary(Input, Unparsed).
+
+
+-spec calculate_parsed_host_port(binary(), binary()) -> binary().
+calculate_parsed_host_port(Input, <<>>) ->
+    strip_last_char(Input, [$:,$?,$#,$/]);
+calculate_parsed_host_port(Input, Unparsed) ->
+    get_parsed_binary(Input, Unparsed).
+
+
+calculate_parsed_query_fragment(Input, <<>>) ->
+    strip_last_char(Input, [$#]);
+calculate_parsed_query_fragment(Input, Unparsed) ->
+    get_parsed_binary(Input, Unparsed).
+
+
+get_port(<<>>) ->
+    undefined;
+get_port(B) ->
+    try binary_to_integer(B)
+    catch
+        error:badarg ->
+            throw({error, invalid_uri, B})
+    end.
+
+
+%% Strip last char if it is in list
+strip_last_char(<<>>, _) -> <<>>;
+strip_last_char(Input, [C0]) ->
+    case binary:last(Input) of
+        C0 ->
+            init_binary(Input);
+        _Else ->
+            Input
+    end;
+strip_last_char(Input, [C0,C1]) ->
+    case binary:last(Input) of
+        C0 ->
+            init_binary(Input);
+        C1 ->
+            init_binary(Input);
+        _Else ->
+            Input
+    end;
+strip_last_char(Input, [C0,C1,C2]) ->
+    case binary:last(Input) of
+        C0 ->
+            init_binary(Input);
+        C1 ->
+            init_binary(Input);
+        C2 ->
+            init_binary(Input);
+        _Else ->
+            Input
+    end;
+strip_last_char(Input, [C0,C1,C2,C3]) ->
+    case binary:last(Input) of
+        C0 ->
+            init_binary(Input);
+        C1 ->
+            init_binary(Input);
+        C2 ->
+            init_binary(Input);
+        C3 ->
+            init_binary(Input);
+        _Else ->
+            Input
+    end.
+
+
+%% Get parsed binary
+get_parsed_binary(Input, Unparsed) ->
+    {First, _} = split_binary(Input, byte_size(Input) - byte_size_exl_head(Unparsed)),
+    First.
+
+
+%% Return all bytes of the binary except the last one. The binary must be non-empty.
+init_binary(B) ->
+    {Init, _} =
+        split_binary(B, byte_size(B) - 1),
+    Init.
+
```

Comment:
I do not understand the why this function is called init_binary? Maybe  binary_strip_last_char?